chain/consensus/mir/state_manager.go

```diff
@@ -285,18 +285,16 @@ func (sm *StateManager) ApplyTXs(txs []*requestpb.Request) error {
 	if err := sm.ctx.Err(); err != nil {
 		return nil
 	}
-	base, err := sm.api.ChainHead(sm.ctx)
+
+	base, err := sm.api.ChainGetTipSetByHeight(sm.ctx, sm.height-1, types.EmptyTSK)
```

Comment:
While this makes sense I don't understand why it makes a difference. Do you mind explaining me why this fixes the issue?

Reply:
Oh, gotcha!!! Nice fix then, folks. This was a hard one.

Reply:
Thank you, @matejpavlovic !